src/server.c

```diff
@@ -1874,6 +1874,7 @@ void createSharedObjects(void) {
     for (j = 0; j < OBJ_SHARED_INTEGERS; j++) {
         shared.integers[j] =
             makeObjectShared(createObject(OBJ_STRING,(void*)(long)j));
+        initObjectLRUOrLFU(shared.integers[j]);
```

Comment:
i'm not sure we need this. in any case when we use LRU / LFU eviction we avoid storing shared objects in the database.
see `createStringObjectFromLongLongWithOptions()`

Reply:
Regardless of whether eviction is turned on or not, we always need to keep the lru/lfu of the key to have a meaningful value, which may help in some scenarios, for example, when judging the idle time of a key in LRU mode

Reply:
with shared objects we can't judge the idle time of a key (the same LRU is shared with multiple keys).
anyway, i don't think this like can cause any damage, just saying it's unnecessary.
you have a point that OBJECT IDLETIME will return garbage on a shared object without this line.
but even with it, it won't return the right result.. i suppose an incorrect value is better than garbage, so let's keep it.

Reply:
but lru of object will not be inited on initObjectLRUOrLFU when o->refcount == OBJ_SHARED_REFCOUNT?